:bug: Fix problem of return PROS_ERR_F error code

Jerrylum opened the pull request:
#### Summary:
Someone is very lazy. Use `claim_port` in all functions.

Modify `claim_port(port, device_type)` to `claim_port(port, device_type, error_code)`
Add claim_port_i(port, device_type) for int32_t function
Add claim_port_f(port, device_type) for double and float function
Modify `claim_port_try`, return 1 if success, else 0(It was return PROS_ERR)

#### Motivation:
Love, Harry, Love.

##### References (optional):
#159 

#### Test Plan:
- [ ] I can't compile the files
- [ ] I don't know what to do.

HotelCalifornia commented:
>  I can't compile the files
>  I don't know what to do.

please come up with a small set of ways that those of us who can compile the code can use verify this change has been made correctly

Jerrylum commented on src/devices/vdml.c:
```diff
@@ -30,17 +30,7 @@ extern void registry_init();
 extern void port_mutex_init();
 
 int32_t claim_port_try(uint8_t port, v5_device_e_t type) {
```
At https://github.com/purduesigbots/pros/blob/develop/include/vdml/vdml.h#L59-L75

Jerrylum replied:
oh... I think I need to update the comment of the claim_port_try.

Jerrylum replied:
Oh no no no... I gave a wrong reference. Look at the PR Commit file. I am true

Jerrylum commented on include/vdml/vdml.h:
```diff
@@ -32,29 +32,50 @@
 
+/**
+ * Function like claim_port. This macro should only be used in functions 
+ * that return int32_t as PROS_ERR could be returned.
```
Got it.

Jerrylum commented:
> some minor things. notably, I'd prefer if you didn't refer explicitly to `true` or `false` when checking the output of `claim_port_try` because it doesn't return a strict `bool` (i.e. from `stdbool.h`), rather it uses the convention that `0` is falsey and nonzero values are truthy.

I pushed a new commit and it should fix what you said. Notice me if any problem.
Also, I am trying to make a Test Plan for you. just wait a minute.

Jerrylum commented:
> you're going to also need to look at src/vdml_adi.c, as the signature for `claim_port` has now changed.

I changed all of them to `claim_port_i` in the `vdml_adi.c`. Also, I had a fully files search in the repo and replaced the signature one by one. So I think it is fine.

HotelCalifornia commented:
> > you're going to also need to look at src/vdml_adi.c, as the signature for `claim_port` has now changed.
> 
> I changed all of them to `claim_port_i` in the `vdml_adi.c`. Also, I had a fully files search in the repo and replaced the signature one by one. So I think it is fine.

Ah yep, so you did. my bad

HotelCalifornia commented:
/AzurePipelines run

HotelCalifornia commented:
/AzurePipelines run

HotelCalifornia commented:
/AzurePipelines run

HotelCalifornia commented on src/devices/vdml.c:
```diff
@@ -30,17 +30,7 @@ extern void registry_init();
-		errno = EACCES;
-		return PROS_ERR;
-	}
+	claim_port(port, type, 0);
```
I know it may be redundant code, but it turns out the difference between `claim_port_try` and `claim_port` is that the latter saves a copy of the device (`v5_smart_device_s_t* device = registry_get_device(port)`) while the former does not. Compiling with `-Wunused-variable` (as we do) complains about this.

HotelCalifornia commented on src/devices/vdml.c:
```diff
@@ -32,14 +32,14 @@ extern void port_mutex_init();
 	}
-	if (registry_validate_binding(port, type) != 0) {
-		return PROS_ERR;
+	if (registry_validate_binding(port, device_type) != 0) {
```
this should still be `type`

Jerrylum replied:
Oh no, I will fix it later.
